### PR TITLE
Shackles and Stone pt1

### PR DESCRIPTION
**Pull request by pisskop**
Adds recently added doodads, namely shackles and Smoothed Stone masonry, to the game's json mapgen files.

  Shackles are by and large limited to bandit camps and other secluded or unsavory places.
  Smoothed stone replaces the majority of the rough stone structures currently using stone.  This includes the evac shelter, as Im under the impression it was masoned stone blocks with a rough exterior as opposed to hunks of rock hauled out of the ground.
  Public buildings or ones traditionally made of impressive stone work were given a ~50% chance to spawn with stone exteriors instead of their inferior wooden ones, although they keep their wooden interior walls.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
While you're adding shackles-related changes, you could also remove "ALLOW_FIELD_EFFECT" from shackles, gravestones and slabs. This flag only makes sense when used with "SEALED" flag, otherwise it does nothing.

File: data/json/mapgen/museum.json [hunk 1]
```diff
@@ -15,7 +15,128 @@
...
+                { "item": "museum_guns", "chance": 45, "x": [ 9,11 ], "y": [ 16,16 ] },
+                { "item": "museum_armor", "chance": 40, "x": [ 4,7 ], "y": [ 19,20 ] },
+                { "item": "museum_melee", "chance": 40, "x": [ 4,7 ], "y": [ 19,20 ] },
+                { "item": "museum_guns", "chance": 40, "x": [ 4,7 ], "y": [ 19,20 ] },
+                { "item": "museum_misc", "chance": 50, "x": [ 4,7 ], "y": [ 19,20 ] },
+                { "item": "museum_security", "chance": 60, "x": [ 2,6 ], "y": [ 13,13 ] },
+                { "item": "museum_security", "chance": 60, "x": [ 2,2 ], "y": [ 14,16 ] }
+                            ]
+}
+}
```
Comment (Coolthulhu) [inline data/json/mapgen/museum.json]: Indents

File: data/json/mapgen/pizza_parlor.json [hunk 2]
```diff
@@ -146,34 +146,140 @@
...
+                "---+-..........GccccBccR",
+                "-sll-..BB..BB..-cccBTBcR",
                 "-lll-..TT..TT..gccccBccR",
                 "-tll-..BB..BB.PgcccccccR",
-                "-||||||||||||||-rfrrrrrR"
+                "----------------rfrrrrrR"
+            ],
+            "terrain": {
+                "-": "t_wall",
+                "|": "t_rock_smooth",
```
Comment (Coolthulhu) [inline data/json/mapgen/pizza_parlor.json]: Some walls are smooth rock and some are the regular walls?

**Comment by pisskop** [comment]
no?

those are baby Ls

File: data/json/mapgen/library1.json [hunk 3]
```diff
@@ -30,27 +30,109 @@
...
+                    "*--------O--O---O---O--*",
+                    "************************",
+                    "************************"
+            ],
+            "terrain": {
+                ".": "t_floor",
+                "*": "t_grass",
+                "~": "t_sidewalk",
+                "-": "t_wall",
+                "|": "t_rock_smooth",
```
Comment (Coolthulhu) [inline data/json/mapgen/library1.json]: This isn't actually used anywhere in the json above.

File: data/json/mapgen/museum.json [hunk 4]
```diff
@@ -15,7 +15,128 @@
...
+            ".-G    - DDD          -.",
+            ".------------L---------.",
+            ".-        -           -.",
+            ".-  ####  -           -.",
+            ".-  ####  +   ccccc   -.",
+            ".-        -   ttttt   -.",
+            ".----------------------.",
+            "........................"
+            ],
+            "terrain": {
```
Comment (Coolthulhu) [inline data/json/mapgen/museum.json]: A lot of redundant entries below this line.

File: data/json/mapgen/pizza_parlor.json [hunk 5]
```diff
@@ -146,34 +146,140 @@
...
+                "F": "f_fridge",
+                "H": "f_chair",
+                "L": "f_locker",
+                "U": "f_chair"
+            },
+            "toilets": {
+                "t": { }
+            },
+            "//": "Default terrain is floor but some things are outside (tables)",
+            "set": [
```
Comment (Coolthulhu) [inline data/json/mapgen/pizza_parlor.json]: Why use `set` instead of including it in terrain settings above?

File: data/json/mapgen/museum.json [hunk 6]
```diff
@@ -15,7 +15,128 @@
...
+            ":": "f_dresser",
+            "T": "f_toilet",
+            "l": "f_vending_c",
+            "S": "f_null",
+            "G": "f_desk",
+            "f": "f_glass_fridge",
+            "5": "f_null",
+            "8": "f_null",
+            "D": "f_displaycase"
+            },
```
Comment (Coolthulhu) [inline data/json/mapgen/museum.json]: Indentation

**Comment by Coolthulhu** [comment]
No errors here, just quality issues.

**Comment by pisskop** [comment]
Im not sure I understand.  I didnt do anything but put  in a chance for rockwalls to spawn.

the flags used are the original.  Im not aware of a more efficient way to add a stone building via json

**Comment by Coolthulhu** [comment]
You copied a large blocks of redundant entries for the mappings.
The mapping in each mapgen should only include actually used symbols. If you're not using the "|" or "1" or "l" or whatever symbol in a given mapgen, remove it from said mapgen.

This isn't currently enforced, but may later start being.

**Comment by pisskop** [comment]
> Why use set instead of including it in terrain settings above?

Because its outside and fill_ter is floor.

**Comment by pisskop** [comment]
Your problems with this PR seem to be with the original documents rather than any of my own additions.  Perhaps a seperate PR is needed to reformat the offending mapgen files, as Im sure there are some that I have not added anything to.

**Comment by Coolthulhu** [comment]
> Because its outside and fill_ter is floor.
> pedit: And also its scattered over an area as opposed to spawning on one tile.

That still doesn't explain why is it set with "set" instead of adding it to mapping and "drawing" it like the rest of the mapgen.

> Your problems with this PR seem to be with the original documents rather than any of my own additions.

Nope. The diffs clearly show you adding the changes I'm talking about. Even when you're just copying and pasting, you should generally clean it up a bit.

Look at the "files changed" of this PR. A green line/block that doesn't have a red line/block paired with it is a new addition.

If you were only cleaning up existing PRs, it would be fine to leave them a bit untidy if they were like that at first. But you're adding totally new entries and that means you should clean them up before they go in.

**Comment by pisskop** [comment]
A PR specifically designed to address the formatting problems you percieve is 'okay' if its untidy, but a PR that is only tangentally related to the problem isnt?  :\

No, this PR did do some cleaning up of the base code, and then some more at your behest.  It is not this PRs job to clean up these specific files to your satisfaction, and it will not do so.

I dont know the percise difference between the 'remap' and 'set' code, or when it would be advisable to use one or the other or why you would insist on conformity when we have 2 valid code styles that both produce the desired result.  I am much less familiar with the remap doodad than the set, and if you were enforcing a syntax than new buildings such as the mortuary would not share the same issues that these older buildings do.

Im willing to do a seperate PR and clean up these files if I am being asked to, but it is not this PRs job.

**Comment by Coolthulhu** [comment]
> A PR specifically designed to address the formatting problems you percieve is 'okay' if its untidy, but a PR that is only tangentally related to the problem isnt?

You added untidy jsons. They weren't there before. You wrote or copied entries that aren't needed. This is your job and your responsibility, because you added them.
Specifically the museum.json file. Below my comment.

Do I have to rephrase it once again?

**Comment by pisskop** [comment]
Suit yourself.

The museum was already cleaned up, and the set code is neither incorrect, in need of immediate remedy, nor inferior so far as I can tell.

**Pull request closed without merging**